Make MiniYaml inherits and removal more flexible

RoosterDragon opened the pull request:
- Previously the Inherits syntax was only resolved when used for top-level nodes. Now it is also resolved for nested nodes as well.
- Previously the MiniYAML Merge feature supported the ability to remove nodes, but this only worked within the context of inherited nodes. Now, we allow node removal to work outside of the inheritance context.
- Fix the sprite issues in d2k, as the old `-Remap` syntax is now considered invalid (before it was valid, but was a no-op and not working as intended)

Fixes #19580 Fixes #16278 Fixes #21444

RoosterDragon committed:
Make MiniYaml inherits and removal more flexible

- Previously the Inherits syntax was only resolved when used for top-level nodes. Now it is also resolved for nested nodes as well.
- Previously the MiniYAML Merge feature supported the ability to remove nodes, but this only worked within the context of inherited nodes. Now, we allow node removal to work outside of the inheritance context.

pchote commented:
Related: #21444.

RoosterDragon commented:
I've realised that as well as making removals more flexible, this also tackles an open request r.e. making Inherits more flexible. Have pushed an extra test case to cover that and updated the description.

PunkPun commented on mods/d2k/sequences/infantry.yaml:
```diff
@@ -83,7 +83,6 @@ light_inf:
 		Filename: DATA.R16
 		Start: 4272
 		Offset: -30,-24
-		-Remap:
```
So change this to `Remap: 00000000` instead to fix #21444?

RoosterDragon replied:
Sure, pushed a separate commit. It does indeed fix the issue.

PunkPun reviewed (commented):
Otherwise LGTM

RoosterDragon committed:
Fix the remap for sprites in d2k.

Follow-up from 34ff23d0305702cabc6ba24bdb8d735ea6317458, the previous `-Remap:` line was not working and has since been removed. Without the remap being unapplied, there are rendering artefacts introduced as a result. Fix this by explicitly setting the remap to something else instead.

PunkPun commented:
[changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/f7fe4efe6c497b86eacdf5c1e4f23b8f7d7be9b5...a15ece3c1f5241471d88ec2f3a480e384aea458f)
